Improved AA test, added more representive results.

tikhomirovd opened the pull request.

VasilievaKA commented on hypex/ab_test/ab_tester.py:
```diff
@@ -273,7 +271,7 @@ def _postprep_data(data, spit_indexes: Dict = None) -> pd.DataFrame:
             spit_indexes: Dict of indexes with separation on test and control group
```
typing for data

VasilievaKA commented on hypex/ab_test/ab_tester.py:
```diff
@@ -305,7 +303,7 @@ def calc_ab_delta(a_mean: float, b_mean: float, mode: str = "percentile"):
 
     def sampling_metrics(
-        self, data: pd.DataFrame, random_state: int = None, test_size: float = 0.5
+            self, data: pd.DataFrame, random_state: int = None, test_size: float = 0.5
```
Missing parameter test_size in docstring

VasilievaKA commented on hypex/ab_test/ab_tester.py:
```diff
@@ -346,15 +344,15 @@ def sampling_metrics(
             ).pvalue
             t_result[f"{tf} ks-test p-value"] = ks_2samp(ta, tb).pvalue
```
there are a lot of repetitions in metric calculations 
it may be worth making a separate function for this

VasilievaKA commented on hypex/ab_test/ab_tester.py:
```diff
@@ -479,14 +476,14 @@ def calc_uniform_tests(
             return pd.DataFrame(results), data_from_sampling
```
previous lines contains too much if-else

VasilievaKA commented on hypex/ab_test/ab_tester.py:
```diff
@@ -548,11 +545,11 @@ def uniform_tests_interpretation(
         return self.aa_score(experiment_results)
 
     def num_feature_uniform_analysis(
```
num_feature_uniform_analysis may be static or out of class

VasilievaKA commented on hypex/ab_test/ab_tester.py:
```diff
@@ -652,7 +649,7 @@ def num_feature_uniform_analysis(
         plt.show()
 
     def cat_feature_uniform_analysis(
```
cat_feature_uniform_analysis may be static or out of class

VasilievaKA commented on hypex/ab_test/ab_tester.py:
```diff
@@ -694,7 +691,8 @@ def split_analysis(self, splited_data: pd.DataFrame, **kwargs):
+                **kwargs):
         labeling = self.columns_labeling(data)
         best_results, best_split = None, None
 
```
line 707 i_combinAtions

VasilievaKA commented on hypex/ab_test/ab_tester.py:
```diff
@@ -745,9 +743,9 @@ def process(self, data: pd.DataFrame, optimize_groups: bool = False, iterations:
 
```
line 735 
what is the usage of variable aa_scores? 
ou don't use it after initialization

VasilievaKA commented on hypex/ab_test/ab_tester.py:
```diff
@@ -228,18 +207,15 @@ def sampling_metrics(
-        # prep data to show user (merge indexes and init data)
-        data_from_sampling_dict[random_state] = self._postprep_data(data, split)
-
+        data_from_sampling_dict = {random_state: self._postprep_data(data, split)}
```
what is the usege of this variable?

tikhomirovd commented on examples/tutorials/Tutorial_2_ABtesting.ipynb:
Tutorial 3 :)

tikhomirovd commented on hypex/ab_test/__init__.py:
```diff
@@ -1,3 +1,3 @@
 
-__all__ = ["AATest", "ABTest", "merge_groups"]
\ No newline at end of file
+__all__ = ["AATest", "ABTest", "merge_groups"]
```
Please, add blank line at the end

VasilievaKA reviewed (dismissed):
good job

tikhomirovd reviewed (commented):
Done

tikhomirovd commented:
## New Features
- **MDE Calculation Function**: Added a function for calculating the Minimum Detectable Effect (MDE).
- **Sample Size Calculation with MDE**: Introduced a function to calculate the sample size required for a given MDE.
- **Test Power Calculation Function**: Implemented a function for calculating the power of the test.

## Enhancements to AA Tests
- **Simplified Pipeline**: The classical pipeline can now be invoked with a single `process` function.
- **Stratification Optimization**: The process now allows for optimization in stratification.
- **Enhanced Test Interpretation**:
  - Built a table with test statistics for better analysis.
  - Created a summary table for AA test outcomes.
  - Developed visualizations (graphs) for AA test results.
- **Improved Split Analysis**:
  - Added graphs for distribution analysis to ensure group homogeneity.
  - Conducted statistical tests for homogeneity.
  - Calculated and presented group statistics.
  - Receiving a brief summary of the test


## General Improvements
- **Bug Fixes**: Addressed and resolved known issues.
- **Code Refactoring**: Improved code structure for better maintainability and readability.
- **Process Optimization**: Enhanced overall process efficiency.
